applets/brightness/brightness-applet.c

```diff
@@ -702,11 +705,6 @@ gpm_applet_popup_cb (GpmBrightnessApplet *applet, GdkEventButton *event)
 		return TRUE;
 	}
 
-	/* don't show the popup if brightness is unavailable */
-	if (applet->level == -1) {
-		return FALSE;
-	}
-
```

Comment:
You removing this condition because it doesn't work?

Reply:
It does in fact seem to fail: the popup slider comes up just fine with the applet on a desktop. Note however that this is can be useful in working on the applet, reducing the number of times the work has to be moved to a laptop. It's the entire applet not just the slider that has no function (outside being worked on) on a desktop, not just the slider.  Sound applet or status icon with no system sound support is a similar case though, and we use a null output there to allow the applet to render anyway and most importantly not crash. This helps users debug a lack of sound support, giving a shortcut to the menu which lists any options the system does support.